### PR TITLE
Custom message should be required if deprecating package due to "Other" reason

### PR DESCRIPTION
**Pull request by scottbommarito**
Part of https://github.com/NuGet/Engineering/issues/2382

image

### CODE REVIEW

**Comment by scottbommarito** [comment]
@anangaur - to clarify, do you still want custom message required if "other" is selected but legacy/bugs is also selected?

**Comment by anangaur** [comment]
Anytime `Other` is selected, the custom message should become mandatory IMO. In other cases, the header itself should have **`Provide custom message `**`(optional)`. On selection of `Other`, it should become: **`Provide custom message `**

File: src/NuGetGallery/Views/Packages/_ManageDeprecation.cshtml [hunk 1]
```diff
@@ -77,13 +77,16 @@
             <label for="customMessage" class="deprecation-section-header">Provide custom message</label>
             <textarea id="customMessage" name="customMessage" class="form-control full-line" data-bind="textInput: customMessage" rows="2"></textarea>
         </div>
+        <div data-bind="visible: requiresCustomMessage()">
+            @ViewHelpers.AlertDanger(@<text>You must add a custom message if you are deprecating a package due to another reason!</text>)
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/_ManageDeprecation.cshtml]: >You must add a custom message if you are deprecating a package due to another reason! [](start = 44, length = 85)

What do you think of the message:

`A custom message is required if <b>other</b> is selected`

It makes it clearer which field added this requirement.

File: tests/NuGetGallery.Facts/Controllers/ManageDeprecationJsonApiControllerFacts.cs [hunk 2]
```diff
@@ -108,6 +108,20 @@ public void ReturnsAllAvailableVersionsInReverseVersionOrder()
...
         {
+            [Fact]
+            public async Task ReturnsBadRequestIfOtherAndNoCustomMessage()
+            {
+                // Arrange
+                var controller = GetController<ManageDeprecationJsonApiController>();
+
+                // Act
+                var result = await controller.Deprecate(
+                    "id", null, false, false, true, null, null, null);
```
Comment (loic-sharma) [inline tests/NuGetGallery.Facts/Controllers/ManageDeprecationJsonApiControllerFacts.cs]: You can use named properties here, right? If so, please add those in here :)

File: src/NuGetGallery/Views/Packages/_ManageDeprecation.cshtml (hunk 1 above)
Comment (scottbommarito) [reply src/NuGetGallery/Views/Packages/_ManageDeprecation.cshtml]: agreed.

**Comment by scottbommarito** [comment]
@anangaur - Agreed with requiring it whenever `other` is selected. I don't think `(optional)` is necessary...the message below when `other` is selected is pretty explicit

File: tests/NuGetGallery.Facts/Controllers/ManageDeprecationJsonApiControllerFacts.cs (hunk 2 above)
Comment (scottbommarito) [reply tests/NuGetGallery.Facts/Controllers/ManageDeprecationJsonApiControllerFacts.cs]: will do--i think this will improve test resiliency

**Comment by anangaur** [comment]
LGTM. Can the “other” option be just “Other”? I guess that’s much ore easier to read.

**Pull request merged**